### ISSUE TITLE
An online community for Iterate Hackerspace?

### ISSUE THREAD

**Issue by kajmagnus** [iteratehackerspace/iteratehackerspace.github.io#37]
Here's a Facebook discussion about creating a help-each-other discussion forum for Iterate Hackerspace: https://www.facebook.com/groups/iterate/permalink/652171061620178/

What could be needed? I can create a community, and we can place it at `forum.iteratehackerspace.com`? (it can be moved to elsewhere later if necessary)

Edgar Aroutiounian @fxfactorial  and Edgar Khanzadian @edgarkhanzadyan  you seem to be the ones who update this website the most, maybe you should be admins?

What's needed:

- Create categories. Looking at the courses on the homepage, maybe C++, Javascript and Linux? Maybe *tags* are better, for that (like at StackOverflow). And instead categories could be like: "Programming questions", and "Startups", and "Interviewing for a job", and "Helping non-profits by writing code" ?

- Intro topics that explains what people can do there. The purpose.

What should the purpose be? Answering programming problems? Or maybe that, and also encouraging / helping people to start startups? social enterprises? finding remote jobs? helping non-profits by writing software for them? Maybe some non-profits would be interested in joining the community and write a bit about what they're doing — maybe the students can find interesting (side) projects to work on, that way.

---

For people to join & start "using" the community, I'm thinking they need to hear / read about it a few times. For example, a Facebook post? a link on the Iterate website homepage? Maybe in some cases the teachers who teach at Iterate, would want to join the forum, and answer follow-up questions the students might have, some time later when the teachers have time, via the forum? Then, the teachers might want to tell the students about the community.

(oops this became a bit much text :- P  )

**Comment by fxfactorial** [comment]
Very much appreciate starting this convo. I know I keep going back and forth on the idea of where the forums should be (on iterate hackerspace or on yerevancoder), but now that you wrote this, I think that the scope of forums with questions like: 

```
Answering programming problems? Or maybe that, and also encouraging / helping people to start startups? social enterprises? finding remote jobs? helping non-profits by writing software for them? Maybe some non-profits would be interested in joining the community and write a bit about what they're doing — maybe the students can find interesting (side) projects to work on, that way.
```

This is VERY VERY nice and is much more encompassing and general than iterate hackerspace. There is also a cultural factor, that is, yerevancoder.com is more broad than the hackerspace. So I think it most def should be under forum.yerevancoder.com 

regarding setting up the topics, yes, the ones listed are great to start out with. Once it is up, we will push hard on marketing. 

Broadly speaking: The purpose is to centralize an online tech discussion place to share wisdom and help that is beyond the walled and ephemeral garden of facebook which plagues Armenian tech. 

Categories so far: javascript, Linux, git, startups, funding, job-advice, interviewing, internships, node, local-universities, events, meetups. So not SO tags, but just general forum section subdivisions. 

Intro topics admins can sticky post once forums up.

**Comment by kajmagnus** [comment]
Ok, yerevancoder.com then :- )

Ok, I like that purpose. (I agree about Facebook.)

The categories seems good.

I'll create a community with those categories then, maybe tomorrow or during the weekend. And a DNS CNAME entry is needed too (I'll post again about that), + I'll generate a Let's-Encrypt HTTPS certificate.

Could it be good to think a bit now about marketing?

What I could maybe do: I was working from the coworking spaces Coworking Armenia and Impact Hub Yerevan. There were some startups and non-profits there, who were sometimes building their own mobile app, or would need to create websites, maybe web apps, later. Maybe they would be interested in joining the community? I could post to their Facebook groups.

What other marketing things are there to do? Mentioning the community for students at the different universities? and ... lots of other things I suppose, that I cannot think of right now :- P

**Comment by fxfactorial** [comment]
Excellent, much appreciated. 

About the https, I'm not entirely clear about the relationship of subdomains and cloud flare + https things, so something to keep in mind. (Is a separate https cert even needed?) 

Re marketing, Armenians will believe it when they see it, otherwise folks will be skeptical or just not get the concept. I would prefer that we first get the truly technical communities onboard first, that is, the professional programmers. Then making sure things are shiny and working, we can target the broader tech community folks.

**Comment by kajmagnus** [comment]
Hi @fxfactorial I created a site here: https://yerevancoder.talkyard.net/ — if you sign up (with Gmail for example) I'll add you as admin.

Yerevancoder.com uses CloudFlare's DNS, right? Should we (you) then add a DNS entry for `forum.yerevancoder.com`? or `community.yerevancoder.com`? that points to `yerevancoder.talkyard.net`?

Actually having CloudFlare proxy the requests, shouldn't be needed in Talkyard's case. Talkyard already uses a CDN for static assets (namely KeyCDN). And sending the remaining non-cacheable requests via CloudFlare's proxy server makes things a tiny bit slower I would think (also, not impossible that something breaks, if CloudFlare inspects and rewrites the HTML to "optimize" it).

Apparently CloudFlare's proxying of requests can be switched off, by toggling something to a "gray cloud". From the docs: (https://support.cloudflare.com/hc/en-us/articles/200168756-How-do-I-add-a-subdomain-to-my-site-)

> All other entries that are not web traffic (mail, ftp, etc.) should be toggled to a gray cloud. A gray cloud means that the traffic is not passing through the Cloudflare network

I'm thinking the main benefit with using CloudFlare's proxying, would be to hide the Talkyard's server IP and protect against DoS attacks. ... But Talkyard already uses Google Cloud Engine which supposedly also protects against DoS attacks.

(Maybe I've been a bit slow with replying — I've been working with privacy related stuff, because of a new law in Europe: General Data Protection Regulation (GDPR), which means people need to be able to download their own personal data, delete their accounts & personal data and some other stuff. Which I think are good things b.t.w. :- ))

**Comment by fxfactorial** [comment]
Added to cloudflare but got these issues 
<img width="971" alt="screen shot 2018-03-26 at 9 34 32 am" src="https://user-images.githubusercontent.com/3036816/37919685-7b2c8ee4-30d9-11e8-9078-6dcaad17439c.png">
<img width="596" alt="screen shot 2018-03-26 at 9 34 57 am" src="https://user-images.githubusercontent.com/3036816/37919686-7b44679e-30d9-11e8-8dbf-4b57486df913.png">

Forums look great and I made my own account as well, under `algebr`. 

Thank you so much for doing this! (And that GDPR seem like a coding nightmare of corner cases)

**Comment by kajmagnus** [comment]
@fxfactorial Ok I needed to add the new address to the forum, i.e. `forum.yerevancoder.com`, so the server knows it's supposed to map that address to the new forum. I've done that & made you an admin.

Creating categories is the next step then? What was the categories list now again:

- Javascript
- Node
- Linux
- Git
- Startups
- Funding
- Job Advice
- Interviewing
- Internships
- Local Universities
- Events

Hmm I'm thinking this is a bit too many categories. There's the magic number 7 plus minus 2: People have problems with lists with more than about 7 things, and the fewer, the better (people are more likely to find the right category).  ... Also there's C, Elexir, Rust, ... Machine learning and Python ... and others, the list of categories could grow a lot.

Maybe these categories:

- Programming  (incl Javascript, Node, HTML, CSS, C++ etc)
- Linux, Git, tools
- Startups   (incl funding)
- Job advice  (incl interviewing, finding a job / internship, and cooperating with others at the workplace?)
- Events

(What is the Local Universities category for?)

**Comment by fxfactorial** [comment]
I like the folded categories. 

Local Universities is so that ppl can talk about their local CS departments, gossip, which teacher is good/bad.

**Comment by kajmagnus** [comment]
Ok, I've created categories now: https://forum.yerevancoder.com/categories

I wrote descriptions for  Programming and Tools (feel free to edit & completely rewrite of course :- )), ... I suppose that for the other categories, we should clarify that it's about startups, and job advice, etc, related to Armenia?

What's the next step? Maybe adding a Forum link to the Iterate homepage? Hmm probably people won't notice it, or ... maybe one person per week would, and s/he goes to the forum, notices it's mostly empty and leaves :- P

I wonder if there're some discussion topics that are interesting enough for people to want to joint & read, or post their thoughts and ideas. Maybe summaries of lectures at Iterate?
